gcp_variant_transforms/libs/infer_headers_util.py

```diff
@@ -0,0 +1,360 @@
+    return vcf_header_io.VcfHeaderFieldTypeConstants.FLOAT
+  else:
+    return vcf_header_io.VcfHeaderFieldTypeConstants.STRING
+
+def _can_cast_to(value, cast_type):
+  """Returns true if `value` can be casted to type `type`"""
+  try:
+    _ = cast_type(value)
+    return True
+  except ValueError:
+    return False
+
+def _get_corrected_type(defined_type, value):
+  # type: (str, Any) -> str
+  """Returns the corrected type according to `defined_type` and `value`.
+
+  It handles one special case PyVCF cannot handle, i.e., the defined type is
+  `Integer`, but the provided value is float. In this case, correct the type
+  to be `Float`.
+
+  Note that if `value` is a float instance but with an integer value
+  (e.g. 2.0), the type will stay the same as `defined_type`.
+  """
+  if defined_type == _HeaderTypeConstants.INTEGER:
+    if isinstance(value, float) and not value.is_integer():
+      return _HeaderTypeConstants.FLOAT
+    if isinstance(value, list):
+      for item in value:
+        corrected_type = _get_corrected_type(defined_type, item)
+        if corrected_type != defined_type:
+          return corrected_type
+  return defined_type
+
+def _infer_mismatched_info_field(field_key,  # type: str
+                                 field_value,  # type: Any
+                                 defined_header,  # type: Dict
+                                 num_alternate_bases  # type: int
+                                ):
+  # type: (...) -> Optional[Info]
+  """Returns corrected info if there are mismatches.
+
+  Two mismatches are handled:
+  - Defined num is `A`, but the provided values do not have the same
+    cardinality as the alternate bases. Correct the num to be `None`.
+  - Defined type is `Integer`, but the provided value is float. Correct the
+    type to be `Float`.
+  Args:
+    field_key: the info field key.
+    field_value: the value of the field key given in the variant.
+    defined_header: The definition of `field_key` in the header.
+    num_alternate_bases: number of the alternate bases.
+  Returns:
+    Corrected info definition if there are mismatches.
+  """
+  corrected_num = defined_header.get(_HeaderKeyConstants.NUM)
+  if (corrected_num == field_counts[_FIELD_COUNT_ALTERNATE_ALLELE] and
+      len(field_value) != num_alternate_bases):
+    corrected_num = field_counts['.']
+
+  corrected_type = _get_corrected_type(
+      defined_header.get(_HeaderKeyConstants.TYPE), field_value)
+
+  if (corrected_type != defined_header.get(_HeaderKeyConstants.TYPE) or
+      corrected_num != defined_header.get(_HeaderKeyConstants.NUM)):
+    return Info(field_key,
+                corrected_num,
+                corrected_type,
+                defined_header.get(_HeaderKeyConstants.DESC),
+                defined_header.get(_HeaderKeyConstants.SOURCE),
+                defined_header.get(_HeaderKeyConstants.VERSION))
+  return None
+
+def _infer_mismatched_format_field(field_key,  # type: str
```

Comment:
I feel the parameters can fit into one line. It will be easier to read if them are in one line.